### PR TITLE
[QI2-1140] add example notebooks

### PR DESCRIPTION
**Pull request by napocornejo**

### CODE REVIEW

File: docs/notebooks/Super Dense Coding.ipynb [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def superdense_encoding_circuit(bit_a: int =0, bit_b: int =0) -> QuantumCircuit:\n",
+    "    circuit = QuantumCircuit(2, 2)\n",
+    "    qubit_a = 0\n",
+    "    qubit_b = 1\n",
+    "    #circuit.reset(qubit_a)\n",
```
Comment (Mythir) [inline docs/notebooks/Super Dense Coding.ipynb:88]: reset works on the new qxelarator. So maybe uncomment?

File: docs/notebooks/Super Dense Coding.ipynb [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    "|   Z\t   | 10 |\n",
+    "|   ZX   | 11 |\n",
+    "\n",
+    "A different Bell state will be encoded for each gate that Bob applies to the qubit. After decoding, each Bell state will result in a different 2-bit message. Remember that the Identity gate doesn't alter the state, and the X and Z gates make the qubit do a pi-rotation over their respective axes.\n",
+    "\n",
+    "The different possible states after encoding will then be:\n",
+    "\n",
+    "After the encoding step, Bobs sends the qubit back to Alice. To decode the message, Alice applies a CNOT-gate followed by a H-gate, see Figure 1. The decoding of the '11' message is: H CNOT (1/sqrt(2)(-10 + 01) = 11\n",
+    "\n",
+    "The states after the different respective gate operations are listed in Table 1."
```
Comment (Mythir) [inline docs/notebooks/Super Dense Coding.ipynb:67]: Which table?

File: docs/notebooks/Super Dense Coding.ipynb [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    "| Gates\t | Classical Message |\n",
+    "|:------:| :------: |\n",
+    "|   I\t   | 00 |\n",
+    "|   X\t   | 01 |\n",
+    "|   Z\t   | 10 |\n",
+    "|   ZX   | 11 |\n",
+    "\n",
+    "A different Bell state will be encoded for each gate that Bob applies to the qubit. After decoding, each Bell state will result in a different 2-bit message. Remember that the Identity gate doesn't alter the state, and the X and Z gates make the qubit do a pi-rotation over their respective axes.\n",
+    "\n",
+    "The different possible states after encoding will then be:\n",
```
Comment (Mythir) [inline docs/notebooks/Super Dense Coding.ipynb:63]: ?

File: docs/notebooks/Super Dense Coding.ipynb [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    "\n",
+    "After the encoding step, Bobs sends the qubit back to Alice. To decode the message, Alice applies a CNOT-gate followed by a H-gate, see Figure 1. The decoding of the '11' message is: H CNOT (1/sqrt(2)(-10 + 01) = 11\n",
```
Comment (Mythir) [inline docs/notebooks/Super Dense Coding.ipynb:65]: Figure 1?

File: docs/notebooks/Super Dense Coding.ipynb [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "### Generate the circuit\n",
+    "\n",
+    "The two qubits are initially prepared in the ground state 00. Before sending the qubit to Bob, Alice needs to prepare an entangled state. So, Alice applies a H-gate followed by a CNOT-gate to obtain the desired entangled state (1/sqrt(2)(00 + 11). Now the qubit is ready to be sent to Bob, who encodes the two classical bits of information.\n",
```
Comment (Mythir) [inline docs/notebooks/Super Dense Coding.ipynb:50]: (1/sqrt(2)(00 + 11) seems to be missing a parenthesis

File: docs/notebooks/Super Dense Coding.ipynb [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+    "        circuit.y(qubit_a)\n",
+    "\n",
+    "    circuit.cz(qubit_b, qubit_a)\n",
+    "    circuit.ry(-1.57079632679, qubit_a)\n",
+    "    circuit.ry(1.57079632679, qubit_b)\n",
+    "\n",
+    "\n",
+    "    circuit.measure(qubit_b, qubit_a)  # in qiskit this measures in z basis\n",
+    "    circuit.measure(qubit_b, qubit_a)  # in qiskit this measures in z basis\n",
+    "    circuit.measure(qubit_b, qubit_a)  # in qiskit this measures in z basis\n",
```
Comment (Mythir) [inline docs/notebooks/Super Dense Coding.ipynb:114]: Three identical measurements?

**Review by Mythir — CHANGES_REQUESTED** [review]

File: docs/notebooks/Super Dense Coding.ipynb (hunk 2 above)
Comment (napocornejo) [reply docs/notebooks/Super Dense Coding.ipynb:67]: Removed, redundant line. It is the table already described before.

File: docs/notebooks/QI2 Performance Test.ipynb [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    "q,qc = pcircuit(4, 1)\n",
+    "qc.draw(output='mpl')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce37a3ec-3a7d-4328-a039-6eb3eda9750a",
+   "metadata": {},
+   "source": [
+    "# Run the cirquit on the Quantum Inspire Platform\n",
```
[the remark this answers is not in the document]
Comment (napocornejo) [reply docs/notebooks/QI2 Performance Test.ipynb:86]: Done.

File: docs/notebooks/QI2 Performance Test.ipynb [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    "First we make a connection to the Quantum Inspire platform and chosing a backend."
```
[the remark this answers is not in the document]
Comment (napocornejo) [reply docs/notebooks/QI2 Performance Test.ipynb:87]: Corrected.

File: docs/notebooks/Super Dense Coding.ipynb (hunk 1 above)
Comment (napocornejo) [reply docs/notebooks/Super Dense Coding.ipynb:88]: Yes.

File: docs/notebooks/Super Dense Coding.ipynb (hunk 4 above)
Comment (napocornejo) [reply docs/notebooks/Super Dense Coding.ipynb:65]: Added.

File: docs/notebooks/Super Dense Coding.ipynb (hunk 5 above)
Comment (napocornejo) [reply docs/notebooks/Super Dense Coding.ipynb:50]: Corrected.

File: docs/notebooks/Super Dense Coding.ipynb (hunk 6 above)
Comment (napocornejo) [reply docs/notebooks/Super Dense Coding.ipynb:114]: Their code adds three measurements for what they call maximum-likelihood 3 readout (ML3 RO), which as I have understood, is reading those three times to improve fidelity of the readout.

Comment (Mythir) [reply docs/notebooks/Super Dense Coding.ipynb:124]: Seems to me like a very backend-specific thing. But if it already was like this in the original notebooks I guess we can leave them as is.

**Review by Mythir — APPROVED** [review]

**Pull request merged**